_security/access-control/document-level-security.md

```diff
@@ -185,3 +185,97 @@ plugins.security.dls.mode: filter-level
+To ensure access is not restricted, you can set the following configuration in  `opensearch.yml`:
+
+```
+plugins.security.dfm_empty_overrides_all: true
+```
+{% include copy.html %}
+
+
+The following examples show what level of access roles with DLS enabled and without DLS enabled, depending on the interaction. These examples can help you decide when to enable the ``plugins.security.dfm_empty_overrides_all` setting.
+
+#### Example: Document access
+
+This example shows that enabling `plugins.security.dfm_empty_overrides_all` is beneficial in scenarios where you need specific users to have unrestricted access to documents, despite being part of a broader group with restricted access.
+
+**Role A with DLS**: This role is granted to a broad group of users and includes Document-Level Security (DLS) to restrict access to specific documents, as shown in the following permission set:
+
+```
+{
+  "index_permissions": [
+    {
+      "index_patterns": ["example-index"],
+      "dls": "[.. some DLS here ..]",
+      "allowed_actions": ["indices:data/read/search"]
+    }
+  ]
+}
+```
+
+**Role B without DLS:** This role is specifically granted to certain users, such as administrators, and does not include DLS, as shown in the following permission set
+
+```
+{
+  "index_permissions" : [
+    {
+      "index_patterns" : ["*"],
+      "allowed_actions" : ["indices:data/read/search"]
+    }
+  ]
+}
+```
+{% include copy.html %}
+
+Setting `plugins.security.dfm_empty_overrides_all` to `true` ensures that administrators assigned Role B can override any document-level security restrictions imposed by Role A. This allows specific users designated in Role B to access all documents, regardless of the restrictions applied by Role A's DLS restrictions.
+
+#### Example: Search template access
+
+In this example, two roles are defined, one with DLS, and another without DLS granting access to search templates:
+
+**Role A with DLS:**
+
+```
+{
+  "index_permissions": [
+    {
+      "index_patterns": [
+        "example-index"
+      ],
+      "dls": "[.. some DLS here ..]",
+      "allowed_actions": [
+        "indices:data/read/search",
+      ]
+    }
+  ]
+}
+```
+{% include copy.html %}
+
+**Role B, without DLS** that grants only access to search templates:
+
+```
+{
+  "index_permissions" : [
+    {
+      "index_patterns" : [ "*" ],
+      "allowed_actions" : [ "indices:data/read/search/template" ]
+    }
+  ]
+}
+```
+{% include copy.html %}
+
+When a user has both Role A and Role B permissions, the query results are filtered based on Role A's DLS, even though Role B doesn't use DLS. The DLS settings are held and the returned access is appropriately restricted. 
+
+With the `plugins.security.dfm_empty_overrides_all` is enabled when a user is assigned both Role A and Role B, Role B's permissions will override Role A's restrictions, allowing that user to access all documents. This ensures that the role without DLS takes precedence in the search query response.
```

Comment:
Would the first sentence be better as "When a user is assigned both Role A and Role B and the `plugins.security.dfm_empty_overrides_all` setting is enabled, Role B's permissions..."?